tensorrt_llm/_torch/auto_deploy/transform/library/quantization.py

```diff
@@ -0,0 +1,257 @@
+        weight_shape = original_weight.shape
+
+        # Quantize the weight
+        new_param = nn.Parameter(
+            quantization_impl.quantize_weight(original_weight), requires_grad=False
+        )
+
+        # Update the parameter in the model
+        modname, _, attrname = param_name.rpartition(".")
+        submod = gm.get_submodule(modname)
+        setattr(submod, attrname, new_param)
+
+        # Register load state dict hook
+        gm._register_load_state_dict_pre_hook(
+            partial(quantization_impl.load_hook, weight_name=param_name)
+        )
+        if quantization_impl.post_load_hook:
+            gm.register_load_state_dict_post_hook(
+                partial(quantization_impl.post_load_hook, weight_name=param_name)
+            )
+
+        # Setup scale names and target module for parameter case
+        def get_scale_name(scale_name):
+            return attrname + "_" + scale_name
+
+        scale_target_module = submod
+        scale_name_prefix = f"{modname}."
+
+    # Weight is a dynamic tensor
+    elif hasattr(weight_node, "meta") and "val" in weight_node.meta:
+        weight_shape = weight_node.meta["val"].shape
+
+        # Create a unique identifier for this dynamic weight node
+        node_id = f"bmm_dynamic_{id(node)}"
+
+        # Setup scale names and target module for dynamic case
+        def get_scale_name(scale_name):
+            return f"{node_id}_{scale_name}"
+
+        scale_target_module = gm  # Register in root module
+        scale_name_prefix = ""
+
+        ad_logger.info(f"Quantized BMM with dynamic weight tensor for node {node}")
+    else:
+        # If we can't determine the shape, skip quantization
+        ad_logger.warning(
+            f"BMM weight is dynamic tensor without shape metadata, skipping quantization for node {node}"
+        )
+        return
+
+    # Common logic for both parameter and dynamic tensor cases
+    # Register scales in the target module
+    for scale_name, scale in quantization_impl.default_scales(weight_shape).items():
+        scale_buffer_name = get_scale_name(scale_name)
+        scale_target_module.register_buffer(scale_buffer_name, scale)
+
+    # Change node target to quantized bmm op
+    node.target = quantization_impl.target_op()
+
+    # Insert scale nodes
+    with gm.graph.inserting_before(node):
+        scales = {}
+        for scale_name in quantization_impl.scale_names():
+            scale_buffer_name = get_scale_name(scale_name)
+            scales[scale_name] = gm.graph.create_node(
+                "get_attr", f"{scale_name_prefix}{scale_buffer_name}"
+            )
+
+    # Update node arguments and kwargs
+    scale_values = [scales[scale_name] for scale_name in quantization_impl.scale_names()]
+    node.args = (*node.args, *scale_values)
+
+
+class QuantizationConfig(TransformConfig):
+    """Configuration for the quantization transform."""
+
+    quant_config: Dict[str, Any] = Field(
```

Comment:
quant_config comes from the model factory. So instead of putting it into the config of the transform, you should be able to retrieve from the factory provided to `_apply`